src/filesystem/implementation.js

```diff
@@ -1869,12 +1869,17 @@ function writeFile(context, path, data, options, callback) {
     return callback(new Errors.EINVAL('flags is not valid', path));
   }
 
-  data = data || '';
```

Comment:
@humphd I moved this line of code so that data = 0 would be handled the same as other numbers.